Documentation/tutorial/intro_rest_api_documentation.md

```diff
@@ -0,0 +1,112 @@
+Introduction to REST API Documentation
+===============================
+
+Introduction
+-------------
+
+DocFX now supports generating documentation from REST APIs following [Swagger specification](http://swagger.io/specification/) version 2.0.
+
+The Swagger RESTful API files *MUST* end with `.swagger.json` or `.swagger2.json` or `_swagger.json` or `_swagger2.json` so that these files are recognized as REST API files in DocFX.
+
+One Swagger API file generates one HTML file. For example. a file `contacts.swagger.json` generates file naming `contacts.swagger.html`.
+
+Basic structure
+--------------
+A single Swagger API file is considered as a unique **File** containing multiple **API**s. The **UID**(Unique IDentifier) for the **Item** is defined as the combination of `host`, `basePath`, `info.title` and `info.version` with `/` as seperator. For example, the following Swagger API file has **UID** equals to `microsoft.com/docfx/Contacts/1.6`:
+
+```json
+{
+  "swagger": "2.0",
+  "info": {
+    "title": "Contacts",
+    "version": "1.6"
+  },
+  "host": "microsoft.com",
+  "basePath": "/docfx",
+  "schemes": [
+    "https"
+  ]
+}
+```
+
+A Swagger API **File** contains multiple **API**s as its children. An **API** is an [Operation Object](http://swagger.io/specification/#operationObject) defined in [Path Item Object](http://swagger.io/specification/#pathItemObject). The **UID**(Unique IDentifier) for this **API** is defined as the combination of the **UID** of the **File** and the `operationId` of the [Operation Object](http://swagger.io/specification/#operationObject). For example, the following `get_contacts` operation has **UID** equal to `microsoft.com/docfx/Contacts/1.6/get_contacts`:
```

Comment:
have a question, when to use **File**? Some is "Swagger API file" and some "Swagger API **File**"?